app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -534,100 +495,102 @@ public void onItemClick(AdapterView<?> listView, View view, int position, long i
             i.putExtra("entity_detail_index", position);
             if (mNoDetailMode) {
                 returnWithResult(i);
-            } else  {
+            } else {
                 startActivityForResult(i, CONFIRM_SELECT);
             }
         }
     }
 
+    public void onBarcodeFetch(String result, Intent intent) {

```

Comment:
This needs an override annotation, as well as all the other implementations of the BarcodeScanListener interface